src/backend/metal/src/native.rs

```diff
@@ -260,7 +261,7 @@ impl hal::DescriptorPool<Backend> for DescriptorPool {
                                 sampler_offset += layout.count;
                                 slice
                                     .iter()
-                                    .map(|s| Some(s.clone()))
+                                    .map(|s| Some(SamplerPtr(s.as_ptr())))
```

Comment:
isn't this just a copy? e.g. ca't we just do `cloned()` instead of `map`

Reply:
It's still `DescriptorSetLayout::Emulated(Vec<pso::DescriptorSetLayoutBinding>, Vec<metal::SamplerState>)`. Do we want `Vec<SamplerPtr>` here instead?

Reply:
I assume we do want `SamplerPtr`, but even then we still need to `map` to get `Some` here

Reply:
oh, nvm then. I thought it's `SamplerPtr`, but it doesn't have to be

Reply:
Let's leave `metal::SamplerState` for now I guess – we appear to rely on it being retained here (not sure if its intentional or if there is supposed to be another owner)